src/types.ts

```diff
@@ -700,4 +700,50 @@ export class StarknetContract {
         const func = <starknet.CairoFunction>this.abi[functionName];
         return adaptOutputUtil(rawResult, func.outputs, this.abi);
     }
+
+    /**
+     * Adapt `Event` to something more readable .
+     * @param eventNames  an array of the event's name that was emitted
+     * @param rawResult array of  the function output as as unparsed space separated string
+     * @returns structured output
+     */
+    adaptEvent(eventNames: string[], rawResult: string[]) {
+        const events: StringMap[] = [];
+        for (let i = 0; i < eventNames.length; i++) {
+            const func = <starknet.CairoEvent>this.abi[eventNames[i]];
+            const adapted = adaptEventUtil(rawResult[i], func.data, this.abi);
+            events.push(adapted);
+        }
+        return events;
+    }
+
+    /**
+     * decode Events to an array with all the arguments emmitted by the events.
+     * @param events array of unstructured events
+     * @returns array of the arguments emmitted by the events
+     */
+    async decodeEvents(events: starknet.Event[]): Promise<StringMap[]> {
```

Comment:
Let's make it return `Promise<DecodedEvent[]>`, where `DecodedEvent` would hold the event name under `name` and could have a property `data` of type `StringMap`.